analysis/pseudobulk-bulk-prediction/model-notebooks/build-assess-models.Rmd

```diff
@@ -60,44 +68,76 @@ Define files to read in:
 pseudobulk_files <- list.files(
   path = file.path(data_dir, "pseudobulk"),
   full.names = TRUE,
-  pattern = "-pseudobulk\\.tsv$"
+  pattern = "_pseudobulk\\.tsv$"
 )
-pseudobulk_names <- stringr::str_split_i(basename(pseudobulk_files), pattern = "-", i = 1)
+pseudobulk_names <- stringr::str_split_i(basename(pseudobulk_files), pattern = "_", i = 1)
 names(pseudobulk_files) <- pseudobulk_names
 
-bulk_counts_file <- file.path(data_dir, "scpca_data", "normalized_bulk_counts.rds")
+bulk_counts_file <- file.path(data_dir, "normalized-bulk-counts.rds")
 bulk_counts <- readr::read_rds(bulk_counts_file)
 
 # Make sure we have the same projects, in the same order
 stopifnot(
   all.equal(names(pseudobulk_files), names(bulk_counts))
 )
+
+# also the tables with fraction expressed
+bulk_frac_expr_df <- readr::read_tsv(
+  file.path(data_dir, "fraction-expressed-bulk.tsv")
+) |>
+  dplyr::rename(bulk_frac_expressed = frac_samples_expressed)
+  
+named_projects <- purrr::set_names(pseudobulk_names, pseudobulk_names)
+expressed_df <- named_projects |>
+  purrr::map(
+    \(project_id) {
+      readr::read_tsv(
+        file.path(data_dir, glue::glue("{project_id}_fraction-expressed-single-cell.tsv"))
+      )
+    }
+  ) |>
+  purrr::list_rbind(names_to = "project_id") |>
+  dplyr::rename(pseudobulk_frac_expressed = frac_samples_expressed) |>
+  dplyr::full_join(bulk_frac_expr_df, by = c("project_id", "ensembl_id"))
 ```
 
 
 ### Functions
 
 This chunk defines a helper function to read and combine bulk and pseudobulk datasets for a single project.
 ```{r}
-prepare_data <- function(pseudo_file, bulk_df, exclude_samples) {
+prepare_data <- function(pseudo_file, bulk_df, id, exclude_samples, expressed_df) {
+  
+  # determine genes to keep
+  keep_genes <- expressed_df |>
+    dplyr::filter(
+      project_id == id, 
+      (pseudobulk_frac_expressed + bulk_frac_expressed) > (params$expr_threshold * 2)
```

Comment:
Since I was saying something about the intent, we might as well put that in the code more directly.
```suggestion
      mean(pseudobulk_frac_expressed, bulk_frac_expressed) > params$expr_threshold
```